keanu-project/src/main/java/io/improbable/keanu/vertices/bool/nonprobabilistic/operators/BoolModelResultVertex.java

```diff
@@ -18,6 +19,7 @@
     private final ModelResult<BooleanTensor> delegate;
 
     public BoolModelResultVertex(ModelVertex model, VertexLabel label) {
+        super(Tensor.SCALAR_SHAPE);
```

Comment:
So we can just do `super()` here for example

Reply:
I thought it was better to force the vertex to declare it's initial shape. Otherwise, when writing another BoolVertex class, you could easily forget the super(shape) call and end up with some unexpected behaviour. Also, since BoolVertex extends Vertex<BooleanTensor>, it will always have an actual shape (from the tensor) and therefore it makes sense to declare it's initial shape. For the other classes that extend directly from Vertex (e.g. generic vertices) they can just rely on the null constructor `super()` because they may not even be tensors.

Reply:
Ok - makes sense